src/main/java/seedu/fast/model/person/PriorityPredicate.java

```diff
@@ -0,0 +1,54 @@
+package seedu.fast.model.person;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.fast.model.tag.PriorityTag;
+import seedu.fast.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code PriorityTag} matches any of the priorities given.
+ */
+public class PriorityPredicate implements Predicate<Person> {
+    public final List<String> tags;
+
+    public PriorityPredicate(List<String> tags) {
+        this.tags = tags;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        Set<Tag> personTags = person.getTags();
+        for (String tag:tags) {
+            for (Tag personTag:personTags) {
```

Comment:
I think the naming scheme here is quite confusing. I understand that your are trying to match the tags belonging to the person to the predicate tag. I think that a better variable name could be `personsTags` rather than `personTags`, and it could be `predicateTag` rather than just `tag`.